six/projects/csm/source/SIXSensorModel.cpp

```diff
@@ -1110,5 +1111,21 @@ double SIXSensorModel::getCorrelationCoefficient(size_t cpGroupIndex,
                          "SIXSensorModel::getCorrelationCoefficient");
     }
 }
+
+DataType SIXSensorModel::getDataType(const csm::Des& des)
+{
+    std::string desid = des.subHeader().substr(NITF_DE_SZ, NITF_DESTAG_SZ);
+    str::trim(desid);
+    const std::string& subheader = des.subHeader();
+
+    const size_t desshsiOffset = 73 + 200;
```

Comment:
So it feels like logically you should be able to either reuse the `nitf_TREDescription` in `XML_DATA_CONTENT.c` and/or construct a `nitf::TRE` with `XML_DATA_CONTENT` and ask it for the field offset and length of this field.  Obviously in some sense the TRE class knows that.

It looks like you could do

```
nitf::TRE tre("XML_DATA_CONTENT");
const size_t desshsiLength = tre.getField("DESSHSI").getLength();
```

But I don't see a good way to get the offset... it would be nice if the `nitf::Field` knew that too.  You could probably iterate through all the TRE fields and add it up until you got there, but then I assume that 200 is the size of the DESubheader upfront so you'd still need that.

It doesn't feel good to have these three magic numbers in here but I don't think we can sanely get rid of all of them.  So let's do this...

- Use a named static constant for the 200 value (something like `DES_SUBHEADER_LENGTH` so you don't have that magic number in two spots
- See if the above code works to get the length of DESSHSI.  If so, use that rather than hardcoding 60
- Leave the 73 in there but with some comment that that's the offset to the start of DESSHSI